app/models/spotlight/custom_field.rb

```diff
@@ -92,7 +93,7 @@ def index_fields_config
     end
 
     def should_generate_new_friendly_id?
-      super || persisted?
+      new_record? && slug.blank?
```

Comment:
We used to update the slug all the time, and the solr field name only if something critical changed... now that we're using the slug for data storage, this should make it more like the field name.